taichi/util/line_appender.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <string>
+
+#include "spdlog/spdlog.h"
+#include "taichi/common/util.h"
+
+TI_NAMESPACE_BEGIN
+
+class LineAppender {
+ public:
+  explicit LineAppender(int indent_size = 2)
+      : single_indent_(indent_size, ' ') {
+  }
+
+  LineAppender(const LineAppender &) = default;
+  LineAppender &operator=(const LineAppender &) = default;
+  LineAppender(LineAppender &&) = default;
+  LineAppender &operator=(LineAppender &&) = default;
+
+  inline const std::string &lines() const {
+    return lines_;
+  }
+
+  template <typename... Args>
+  void append(std::string f, Args &&... args) {
+    lines_ += indent_ + fmt::format(f, std::forward<Args>(args)...) + '\n';
+  }
+
+  inline void append_raw(const std::string &s) {
+    lines_ += s + '\n';
```

Comment:
Do we need `ident_` here?

Reply:
From my current use case, I feel like the answer is no. `s` could contain multiple lines, so even if we add `indent_`, it will only be to the first line.  So i guess the indentation here is messed up anyway, unless we process `s` line by line...

Reply:
I see. That makes sense.